astropy/coordinates/sky_coordinate.py

```diff
@@ -358,6 +355,10 @@ def __getattr__(self, attr):
         Overrides getattr to return coordinates that this can be transformed
         to, based on the alias attr in the master transform graph.
         """
+
+        if self._attr_getters is not None and attr in self._attr_getters:
+            return self._attr_getters[attr]()
```

Comment:
The performance gains from this change are *marginal* in most cases, but in a loop it does make a difference when the same attributes are looked up over and over again.